Mono.Cecil.PE/ImageWriter.cs

```diff
@@ -61,7 +61,8 @@ sealed class ImageWriter : BinaryStreamWriter {
 			if (metadataOnly)
 				return;
 
-			this.pe64 = module.Architecture == TargetArchitecture.AMD64 || module.Architecture == TargetArchitecture.IA64 || module.Architecture == TargetArchitecture.ARM64;
+			this.pe64 = module.Architecture == TargetArchitecture.AMD64 || module.Architecture == TargetArchitecture.IA64 || module.Architecture == TargetArchitecture.ARM64 ||
+				    module.Architecture == TargetArchitecture.RiscV64;
```

Comment:
In System.Reflection.Metadata, we have been emitting PE64 header [only for architectures that are natively supported by Windows](https://github.com/dotnet/runtime/blob/256191f27a95d2ea8d2e9785817a7d1ae50e8505/src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/ManagedTextSection.cs#L93). PE64 header is unnecessary for IL-only files outside Windows.

Would it be better to adopt similar strategy for Cecil - flip non-Windows architectures originating in R2R file to AnyCPU when writing the IL-only file? It would save us from introducing this ever-growing list of 64-bit architectures.

Reply:
Sorry for late response. I just removed the check codes for pe64. As far as I understand, pe64 is not needed to be set to true for riscv-64 because pe64 header is unnecessary for IL-only on non-Windows. Am I right? 

And I cannot understand `flip non-Windows architectures originating in R2R file to AnyCPU when writing the IL-only file?` well. Does it mean that cecil replace arch to AnyCPU on the case (non-Windows architectures originating in R2R file when writing the IL-only file)? If then, it is not a scope which I planned in this PR.

Reply:
> pe64 header is unnecessary for IL-only on non-Windows. Am I right?

Right. pe64 header is only necessary for 64-bit architecture specific images.

> Does it mean that cecil replace arch to AnyCPU on the case

I assume that the problem you are running into is the exception at https://github.com/dotnet/runtime/blob/fa1164c327052042120bf0d3b6fc81e86cfab69d/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs#L79 .

This logic is not quite right (it won't compute the architecture of the IL image that the crossgened image was created from) and too complicated for not good reason. 

It would best to delete this logic and instead change the output for crossgened input binaries to always be AnyCPU here: https://github.com/dotnet/runtime/blob/fa1164c327052042120bf0d3b6fc81e86cfab69d/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs#L128